### PR TITLE
Refactor logging: make structured, remove duplicates, formatting

### PR DESCRIPTION
**Pull request by alebedev87**
Follow-up of the review comments about the logging from https://github.com/openshift/node-observability-operator/pull/56.

### CODE REVIEW

File: pkg/operator/controller/machineconfig/controller.go [hunk 1]
```diff
@@ -265,7 +259,6 @@ func (r *MachineConfigReconciler) removeFinalizer(ctx context.Context, req ctrl.
 
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
 		if err := r.ClientGet(ctx, req.NamespacedName, withoutFinalizers); err != nil {
-			r.Log.Error(err, "failed to fetch nodeobservabilitymachineconfig resource for removing finalizer")
```
Comment (sherine-k) [inline pkg/operator/controller/machineconfig/controller.go:268]: Maybe we could add the following for context: 
```
err = utilerrors.NewAggregate([]error{err, fmt.Errorf("failed to fetch nodeobservabilitymachineconfig resource for removing finalize: %w", err)}
```

File: pkg/operator/controller/machineconfig/controller.go [hunk 2]
```diff
@@ -289,7 +282,6 @@ func (r *MachineConfigReconciler) removeFinalizer(ctx context.Context, req ctrl.
 
 		withoutFinalizers.Finalizers = newFinalizers
 		if err := r.ClientUpdate(ctx, withoutFinalizers); err != nil {
-			r.Log.Error(err, "failed to remove nodeobservabilitymachineconfig resource finalizers")
```
Comment (sherine-k) [inline pkg/operator/controller/machineconfig/controller.go:292]: Maybe we could add the following for context:
```
err = utilerrors.NewAggregate([]error{err, fmt.Errorf("failed to remove nodeobservabilitymachineconfig resource finalizers: %w", err)}

File: pkg/operator/controller/machineconfig/mcp.go [hunk 3]
```diff
@@ -133,15 +133,15 @@ func (r *MachineConfigReconciler) GetProfilingMCP(name string) *mcv1.MachineConf
 func (r *MachineConfigReconciler) createMCP(ctx context.Context, name string) error {
 	mcp := r.GetProfilingMCP(name)
 
-	if err := ctrlutil.SetControllerReference(r.CtrlConfig, mcp, r.Scheme); err != nil {
-		r.Log.Error(err, "failed to update owner info in MCP", "MCP", name)
+	if err := ctrlutil.SetOwnerReference(r.CtrlConfig, mcp, r.Scheme); err != nil {
```
Comment (sherine-k) [inline pkg/operator/controller/machineconfig/mcp.go:136]: I'm having a doubt about this one. I think this was changed to SetControllerReference very recently

**Review by sherine-k — APPROVED** [review]
Thanks so much @alebedev87 for taking the time to make this right. 
There's just one doubt from me about SetOwnerReference instead of SetControllerReference, and we're good to go

File: pkg/operator/controller/machineconfig/mcp.go (hunk 3 above)
Comment (alebedev87) [reply pkg/operator/controller/machineconfig/mcp.go:136]: Yes, my fault, bad merge conflict resolution. `ControllerRef` must be kept to make the cascade deletion and owner re-queuing working.

File: pkg/operator/controller/machineconfig/controller.go (hunk 1 above)
Comment (alebedev87) [reply pkg/operator/controller/machineconfig/controller.go:268]: This message doesn't really bring any new information: 
- if fetching fails it'll be in the error
- the fact that it's done for the removal of the finalizers will be mentioned up in the call stack (`cleanUp` method)

File: pkg/operator/controller/machineconfig/controller.go (hunk 2 above)
Comment (alebedev87) [reply pkg/operator/controller/machineconfig/controller.go:292]: This message contains the duplicate information, the error returned by this method will be wrapped in the error with the same message up the call stack (in `cleanUp` method).

**Comment by thejasn** [comment]
For logging across multiple controllers, we should probably prefix with the controller name for better readability. 
```go
const (
	controllerName = "machineconfig_controller"
)

var log = logf.Logger.WithName(controllerName)

```
And we use this `log` instead of fetching it from context. WDYT?

**Review by sherine-k — APPROVED** [review]

**Pull request merged**